### ISSUE TITLE
Wrong protonation for C 

### ISSUE THREAD

**Issue by kexul** [openbabel/openbabel#2353]
I have a molecule which SMILES is `CC(C)CC(=O)Nc1cc(-c2ccc(C(N)=O)cc2C(=O)O)cc(C2CC(C)(c3ccccc3)c3cc(C(=N)N)ccc3N2)c1`, after add H by openbabel, there is a `C` with `+1`.

- [x] I believe this to be a bug with Open Babel
- [ ] This is a feature request

## Environment Information
Open Babel version: 3.1.1
Operating system and version: win10 and centos7

## Expected Behavior

![image](https://user-images.githubusercontent.com/5920217/114177985-a98de380-996f-11eb-84ed-574ab5b5d8ba.png)

## Actual Behavior

![image](https://user-images.githubusercontent.com/5920217/114178002-b0b4f180-996f-11eb-9a7a-66dabb2e9e89.png)

## Steps to Reproduce

```
obabel -:'CC(C)CC(=O)Nc1cc(-c2ccc(C(N)=O)cc2C(=O)O)cc(C2CC(C)(c3ccccc3)c3cc(C(=N)N)ccc3N2)c1' -O output.sdf -p 7
```

**Comment by mquevill** [comment]
I ran the following commands to try to reproduce your issue, but did not get the positively charged C.
(OpenBabel installed via Conda, `openbabel 3.1.1 py37h200e996_1 conda-forge`)

```sh
obabel -:'CC(C)CC(=O)Nc1cc(-c2ccc(C(N)=O)cc2C(=O)O)cc(C2CC(C)(c3ccccc3)c3cc(C(=N)N)ccc3N2)c1' -O output.sdf -p 7
obabel output.sdf -O out.svg -d -xC
```

Resulting image:
![image](https://user-images.githubusercontent.com/17033543/114439644-3666b600-9b97-11eb-8d96-32536f66c2b7.png)

If I inspect the output.sdf file, I see the lines that have charged atoms (only O- and N+), as well as two C atoms that have undefined stereo center (3). All other atoms have no charge on them.:
```
O   0  5  0  0  0  0  0  0  0  0  0  0
N   0  3  0  0  0  0  0  0  0  0  0  0
C   0  0  3  0  0  0  0  0  0  0  0  0
C   0  0  3  0  0  0  0  0  0  0  0  0
```

**Comment by kexul** [comment]
[output.zip](https://github.com/openbabel/openbabel/files/6300771/output.zip)
Hi, @mquevill , that's strange, this is the `.sdf` I've got with the same command.

**Comment by mquevill** [comment]
There is apparently be a difference between the Conda 3.1.1 version and the released 3.1.1 source. If I checkout the `openbabel-3-1-1` tag on GitHub and build, I can replicate your error exactly (with `obgui` or `obabel`). However, it is fixed in both the Conda-released version and the current `master` branch! [I believe PR #2328 contained changes that fixed this issue.]

If you do not want to build from source or use Conda, you may have to wait for another release.

**Comment by ghutchis** [comment]
We certainly intend on a release in a few weeks.
